Geometry/GEMGeometryBuilder/src/ME0GeometryBuilder.cc

```diff
@@ -359,3 +391,160 @@ ME0GeometryBuilderFromDDD::ME0BoundPlane ME0GeometryBuilderFromDDD::boundPlane(c
+    me0Num.baseNumberToUnitNumber(mbn);
+    ME0DetId detId = ME0DetId(me0Num.getDetId());
+    ME0DetId detIdCh = detId.chamberId();
+
+    // build chamber
+    ME0Chamber* me0Chamber = buildChamber(fv, detIdCh);
+    geometry->add(me0Chamber);
+
+    bool doLayers = fv.nextSibling();
+    // loop over layers of the chamber
+    while (doLayers) {
+      MuonBaseNumber mbn = muonConstants.geoHistoryToBaseNumber(fv.history());
+      cms::ME0NumberingScheme me0Num(muonConstants.values());
+      me0Num.baseNumberToUnitNumber(mbn);
+      ME0DetId detId = ME0DetId(me0Num.getDetId());
+      ME0DetId detIdLa = detId.layerId();
+
+      // build layer
+      ME0Layer* me0Layer = buildLayer(fv, detIdLa);
+      me0Chamber->add(me0Layer);
+      geometry->add(me0Layer);
+
+      fv.down();  // down to the first eta partion
+
+      // build first eta partition
+      MuonBaseNumber mbnbis = muonConstants.geoHistoryToBaseNumber(fv.history());
+      cms::ME0NumberingScheme me0Numbis(muonConstants.values());
+      me0Numbis.baseNumberToUnitNumber(mbnbis);
+      ME0DetId detIdbis = ME0DetId(me0Numbis.getDetId());
+      ME0EtaPartition* etaPart = buildEtaPartition(fv, detIdbis);
+      me0Layer->add(etaPart);
+      geometry->add(etaPart);
+
+      bool doEtaParts = fv.sibling();
+      // loop over the other eta partions
+
+      while (doEtaParts) {
+        MuonBaseNumber mbn = muonConstants.geoHistoryToBaseNumber(fv.history());
+        cms::ME0NumberingScheme me0Num(muonConstants.values());
+        me0Num.baseNumberToUnitNumber(mbn);
+        ME0DetId detId = ME0DetId(me0Num.getDetId());
+        // build other eta partitions
+        ME0EtaPartition* etaPart = buildEtaPartition(fv, detId);
+        me0Layer->add(etaPart);
+        geometry->add(etaPart);
+
+        doEtaParts = fv.sibling();
+      }
+      doLayers = fv.nextSibling();
+    }
+    fv.parent();
+    doChambers = fv.firstChild();
+  }
+
+  return geometry;
+}
+
+ME0Chamber* ME0GeometryBuilder::buildChamber(cms::DDFilteredView& fv, ME0DetId detId) const {
+  std::vector<double> dpar = fv.parameters();
+
+  double L = dpar[3];
+  double T = dpar[2];
+  double b = dpar[0];
+  double B = dpar[1];
+  bool isOdd = false;
+  ME0BoundPlane surf(boundPlane(fv, new TrapezoidalPlaneBounds(b, B, L, T), isOdd));
+  ME0Chamber* chamber = new ME0Chamber(detId.chamberId(), surf);
+
+  return chamber;
+}
+
+ME0Layer* ME0GeometryBuilder::buildLayer(cms::DDFilteredView& fv, ME0DetId detId) const {
+  std::vector<double> dpar = fv.parameters();
+
+  double L = dpar[3];
+  double t = dpar[2];
+  double b = dpar[0];
+  double B = dpar[1];
+  bool isOdd = false;
+  ME0BoundPlane surf(boundPlane(fv, new TrapezoidalPlaneBounds(b, B, L, t), isOdd));
+  ME0Layer* layer = new ME0Layer(detId.layerId(), surf);
+
+  return layer;
+}
+
+ME0EtaPartition* ME0GeometryBuilder::buildEtaPartition(cms::DDFilteredView& fv, ME0DetId detId) const {
+  //      auto nStrips = fv.get<double>("nStrips"); //it doesn't work
```

Comment:
delete commented-out code? or it is there to be fixed?

Reply:
@kpedro88 : yes, it is there to be fixed.